### DISCUSSION TITLE
[PMC: 1-basis C-N] Looking for related formally published scientific literature

### DISCUSSION THREAD

**Discussion by proof-theory** [xamidi/pmGenerator#4] [Q&A] [answered]
I am looking for formally published literature related to the problems discussed in [[PMC] Minimal 1-bases for C-N propositional calculus](https://github.com/xamidi/pmGenerator/discussions/2).  
I noticed that [the mentioned paper](http://fitelson.org/walsh.pdf) by Walsh and Fitelson remains an uncorrected preprint, informally published in 2021.

- Are there published proofs relevant for that challenge which are also styled in a similar manner?
- Where can I find more closely related literature?

**Comment by xamidi** [comment, marked as answer]
AFAIK, none of Walsh's six single axioms for C-N propositional calculus occurred in a peer-reviewed journal (yet)<sup>✻</sup>, but Meredith's single axiom was first mentioned by Carew Meredith in *The Journal of Computing Systems* in 1953, which is [available on the Internet Archive](https://archive.org/details/journal-of-computing-systems_1953-07_1_3/page/157/mode/2up).  
Meredith proved  **m**: L1 – L3<sup>❈</sup> there, but his proof is based on formulas, doesn't restore variable order and contains every utilized substitution, e.g. his first three lines

```
1   CCCCCpqCNrNsrtCCtpCsp
    1 p/Cpq, q/CNCNtNrNs, r/t, s/r, t/CCtpCsp × C1 r/CNtNr - 2
2   CCCCtpCspCpqCrCpq 
```

are just a way to write

```
(m) CCCCCpqCNrNsrtCCtpCsp
1. CCCCCpqCNCNtNrNsCNtNrtCCtpCsp  (m) ; 〈r, CNtNr〉
2. CCCCCCpqCNCNtNrNsCNtNrtCCtpCspCCCCtpCspCpqCrCpq  (m) ; 〈p, Cpq〉, 〈q, CNCNtNrNs〉, 〈r, t〉, 〈s, r〉, 〈t, CCtpCsp〉
3. CCCCtpCspCpqCrCpq  (MP):1,2
```

which is isomorphic to:

```
    CCCCCpqCNrNsrtCCtpCsp = 1
[0] CCCCpqCrqCqsCtCqs = D11
```

<sup>✻</sup><sub>Prof. Fitelson informed me one month ago in an email that their new plan was to “write a new (more comprehensive) paper”.</sub>  
<sup>❈</sup><sub>Provided in order L1,L3,L2, i.e. `CCpqCCqrCpr,CpCNpq,CCNppp` rather than `CCpqCCqrCpr,CCNppp,CpCNpq`.</sub>

Theorems in normal Polish notation are generally well suited for search engines.

<a name="literature"></a>

Regarding more recent literature and a prettier notation, I found the paper [Lemmas: Generation, Selection, Application](https://link.springer.com/chapter/10.1007/978-3-031-43513-3_9) by M. Rawson, C. Wernhard, Z. Zombori and W. Bibel, which has been published as part of Springer's conference proceedings series [Lecture Notes in Computer Science](https://www.journal-finder.org/journal/8645) for [TABLEAUX 2023](https://link.springer.com/conference/tableaux). It has an [extended version](https://arxiv.org/abs/2303.05854) on arXiv, which in appendix F (last page) provides a [constructive proof](https://en.wikipedia.org/wiki/Constructive_proof) with “MGTs” (most general theorems) for TPTP's [`1.00`](https://web.archive.org/web/20240824103145/https://tptp.org/CASC/J12/Design.html#Problems)-rated<sup>✻</sup> [LCL073-1](https://web.archive.org/web/20241212213152/https://tptp.org/cgi-bin/SeeTPTP?Category=Problems&Domain=LCL&File=LCL073-1.p) — the problem to prove that L1 can be derived from Meredith's single axiom via condensed detachment.

Their proof has a similar notation to the here used proof summaries, so it can straightforwardly be translated into

```
    CCCCCpqCNrNsrtCCtpCsp = 1
[0] CCCppqCrq = D1D1D11
[1] CpCqCrr = D[0][0]
[2] CCCpCqqrCsr = D1[1]
[3] CCCpqrCqr = D1[2]
[4] CpCCpqCrq = D[3]1
[5] CpCCCqprCsr = D[3][4]
[6] CCCpqrCCCqsCNrNpr = DDD1D1[5][4]1
[7] CCCpqCNCCCrpsCtsNrCCCrpsCts = D[6][4]
[8] CCCpqCNCrpNCCCsCptuCvuCrp = D[6]D1[7]
[9] CpCCqrCCNqNpr = DD1D1D[2][8]1
[10] CCCpNqCCqrCsrCtCCqrCsr = D1D[4]D1DD1D[7]D[7]DD[9][1][2]1
[11] CCpCqrCCCsCtNprCqr = DDD[10]D[3]D[6][10]1[5]
[12] CCCCCpqrstCCCqrst = D1D[11]D1D[11][3]
[13] CCpqCCqrCpr = DD1D[11]DDDD[11][4][7][9][8]D[12]D[12]1
```

for which `./pmGenerator --transform file.txt -f -n -p -2 -d` reveals that it has a fundamental proof length of 6553 steps, compared to [currently 371 steps](https://github.com/xamidi/pmGenerator/discussions/2#challenge-proofs)<sup>❈</sup> for the smallest known D-proof, which was found via `--transform -x`.  
They also have a big list of references towards related literature, some informally published — including Walsh's and Fitelson's preprint.

Searching for `LCL073-1` further led me to [CD Tools](http://cs.christophwernhard.com/cdtools/), whose SGCD<sup>✾</sup> prover according to the site was only the second program (after OTTER with a [highly customized workflow](https://link.springer.com/article/10.1023/A:1010691726881)) to solve it automatically.


<sup>✻</sup><sub>None of their 92 [proposed](https://web.archive.org/web/20241212223928/https://tptp.org/cgi-bin/SeeTPTP?Category=Solutions&Domain=LCL&File=LCL073-1) automated theorem provers generated an actual solution — surprisingly [not even Vampire 4.9](https://web.archive.org/web/20241212114454/https://tptp.org/cgi-bin/SeeTPTP?Category=Solutions&Domain=LCL&File=LCL073-1&System=Vampire---4.9.TMO-Non.f) — which renders this problem “unsolved”.</sub>  
<sup>❈</sup><sub>Updated to the result from February 26, 2025.</sub>  
<sup>✾</sup><sub>Short for “Structure Generation for Condensed Detachment”, according to [this paper by C. Wernhard](https://arxiv.org/abs/2207.08453) on arXiv.</sub>

**Update:** With TPTP v9.1.0 (released in June 2025), Walsh's six C-N axioms were added to the TPTP Problem Library.  
Together with a new `fof`-defined variant [LCL073+1](https://tptp.org/cgi-bin/SeeTPTP?Category=Problems&Domain=LCL&File=LCL073+1.p) of the previously mentioned `cnf`-defined [LCL073-1](https://web.archive.org/web/20241212213152/https://tptp.org/cgi-bin/SeeTPTP?Category=Problems&Domain=LCL&File=LCL073-1.p), there are now six `fof`-defined problem files to show that L1 can be derived from **w1**, **w2**, …, **w6**, which are [LCL978+1](https://tptp.org/cgi-bin/SeeTPTP?Category=Problems&Domain=LCL&File=LCL978+1.p), [LCL979+1](https://tptp.org/cgi-bin/SeeTPTP?Category=Problems&Domain=LCL&File=LCL979+1.p), [LCL980+1](https://tptp.org/cgi-bin/SeeTPTP?Category=Problems&Domain=LCL&File=LCL980+1.p), [LCL981+1](https://tptp.org/cgi-bin/SeeTPTP?Category=Problems&Domain=LCL&File=LCL981+1.p), [LCL982+1](https://tptp.org/cgi-bin/SeeTPTP?Category=Problems&Domain=LCL&File=LCL982+1.p) and [LCL983+1](https://tptp.org/cgi-bin/SeeTPTP?Category=Problems&Domain=LCL&File=LCL983+1.p) with TPTP v9.1.0 problem difficulty ratings `0.85`, `1.00`, `1.00`, `0.80`, `0.35` and `1.00`, respectively.

**Reply by xamidi** [reply]
Regarding more distantly related literature, there is the option to search by _Mathematics Subject Classification (MSC2020)_, e.g. on [zbMATH](https://zbmath.org/classification/?q=cc%3A03F). I would classify proof minimization as
- [03-XX](https://zbmath.org/classification/?q=cc%3A03) Mathematical logic and foundations
  - [03Fxx](https://zbmath.org/classification/?q=cc%3A03F) Proof theory and constructive mathematics
    - [03F07](https://zbmath.org/classification/?q=cc%3A03F07) Structure of proofs
    - [03F20](https://zbmath.org/classification/?q=cc%3A03F20) Complexity of proofs

and _algorithmic_ proof minimization, i.e. proof compression, additionally as:
- [03-XX](https://zbmath.org/classification/?q=cc%3A03) Mathematical logic and foundations
  - [03-08](https://zbmath.org/classification/?q=cc%3A03-08) Computational methods for problems pertaining to mathematical logic and foundations
- [68-XX](https://zbmath.org/classification/?q=cc%3A68) Computer science
  - [68Vxx](https://zbmath.org/classification/?q=cc%3A68V) Computer science support for mathematical research and practice
    - [68V15](https://zbmath.org/classification/?q=cc%3A68V15) Theorem proving (automated and interactive theorem provers, deduction, resolution, etc.)
  - [68Wxx](https://zbmath.org/classification/?q=cc%3A68W) Algorithms in computer science
    - [68W05](https://zbmath.org/classification/?q=cc%3A68W05) Nonnumerical algorithms
    - [68W30](https://zbmath.org/classification/?q=cc%3A68W30) Symbolic computation and algebraic computation

Exemplary search queries: "[`system cc:03F07 cc:03F20`](https://zbmath.org/?q=system+cc%3A03F07+cc%3A03F20)", "[`cc:03F20 cc:68W30`](https://zbmath.org/?q=cc%3A03F20+cc%3A68W30)", "[`short propositional cc:03F20`](https://zbmath.org/?q=short+propositional+cc%3A03F20)"

It should be noted that there is a lot of literature in proof theory that is **highly unrelated** to the here relevant subjects. For example, the textbook [Proof Theory: The First Step into Impredicativity](https://link.springer.com/book/10.1007/978-3-540-69319-2) by Wolfram Pohlers self-declares to fall into only different subcategories:
- [03-XX](https://zbmath.org/classification/?q=cc%3A03) Mathematical logic and foundations
  - [03Fxx](https://zbmath.org/classification/?q=cc%3A03F) Proof theory and constructive mathematics
    - [03F03](https://zbmath.org/classification/?q=cc%3A03F03) Proof theory in general (including proof-theoretic semantics)
    - [03F05](https://zbmath.org/classification/?q=cc%3A03F05) Cut-elimination and normal-form theorems
    - [03F15](https://zbmath.org/classification/?q=cc%3A03F15) Recursive ordinals and ordinal notations
    - [03F25](https://zbmath.org/classification/?q=cc%3A03F25) Relative consistency and interpretations
    - [03F30](https://zbmath.org/classification/?q=cc%3A03F30) First-order arithmetic and fragments
    - [03F35](https://zbmath.org/classification/?q=cc%3A03F35) Second- and higher-order arithmetic and fragments

<a name="search-queries"></a>More relevant search query results that may also help with finding closely related categories:

- [`"condensed detachment"`](https://zbmath.org/?q=%22condensed+detachment%22) (≥ 36 results)
- [`"proof compression"`](https://zbmath.org/?q=%22proof+compression%22) (≥ 14 results)
- [`"proof minimization"`](https://zbmath.org/?q=%22proof+minimization%22) (≥ 4 results)

Categories specific to the here conducted kind of proof minimization:

- [03-XX](https://zbmath.org/classification/?q=cc%3A03) Mathematical logic and foundations
  - [03Bxx](https://zbmath.org/classification/?q=cc%3A03B) General logic
    - [03B05](https://zbmath.org/classification/?q=cc%3A03B05) Classical propositional logic
  - [03-04](https://zbmath.org/classification/?q=cc%3A03-04) Software, source code, etc. for problems pertaining to mathematical logic and foundations
  - [03-11](https://zbmath.org/classification/?q=cc%3A03-11) Research data for problems pertaining to mathematical logic and foundations
- [68-XX](https://zbmath.org/classification/?q=cc%3A68) Computer science
  - [68Qxx](https://zbmath.org/classification/?q=cc%3A68Q) Theory of computing
    - [68Q17](https://zbmath.org/classification/?q=cc%3A68Q17) Computational difficulty of problems (lower bounds, completeness, difficulty of approximation, etc.)
  - [68Rxx](https://zbmath.org/classification/?q=cc%3A68R) Discrete mathematics in relation to computer science
    - [68R07](https://zbmath.org/classification/?q=cc%3A68R07) Computational aspects of satisfiability
  - [68Wxx](https://zbmath.org/classification/?q=cc%3A68W) Algorithms in computer science
    - [68W10](https://zbmath.org/classification/?q=cc%3A68W10) Parallel algorithms in computer science
    - [68W15](https://zbmath.org/classification/?q=cc%3A68W15) Distributed algorithms
    - [68W32](https://zbmath.org/classification/?q=cc%3A68W32) Algorithms on strings
